### DISCUSSION TITLE
Welcome to Gist of Rust Book Discussions!

### DISCUSSION THREAD

**Discussion by katopz** [gist-rs/book#4] [Announcements]
## 🦀 Welcome Rustacean!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by tommeeeee** [comment]
Dear P'KaTopz,🦀 @katopz 

my name is Nammon 
 
I'm new joiner of rust and I've already make a decision of my blockchain journey to move from ETH base to Solana, it make me feel like a frog in the well which has just seen the outside world hahaha,the god of crab 🦀 ( rust ) is totally open my world, and solana is so much better krub.

Could I help to contribute this repo gist rust book In solana section for more, if i found any research or new update of solana. 

🦀

**Comment by katopz** [comment]
Welcome to the club! Please feel free to contribute and enjoy. 😊

**Comment by tommeeeee** [comment]
Hi P' @katopz  I've some interesting topic to discuss krub, it about solana uptime around 4 hr. the report about this outage explain reason how this happen, it's happen from ```BPF loader``` it's seem like come from an update from solana Improvement Proposal that impact to BPF, BPF can use metadata https://github.com/solana-labs/solana/pull/35093, 

you can see major outage in 6 feb, that take around 4 - 5 hr, https://status.solana.com/uptime/qxs6x3yc9cxg?page=1

In my understand this update will not take an impact if all validator not update for a bug version.
but I can happen cause when the bug node take a turn to be a leader node to generate block, so it can't process the new block that why all the network has outage right ? 

so what you think about this outage krub, or you have any idea to share in you perspective about this outage krub.

Ref :
- https://twitter.com/matthew_sigel/status/1754874183807307828
- https://coingape.com/solana-down-whats-the-reason-behind-recurrent-outages/#:~:text=Solana%20Missed%20100%25%20Uptime,-Solana%20network%20has&text=The%20network%20suffered%20a%20major,sentiment%20on%20Solana%20and%20prices.

**Reply by katopz** [reply]
Far as I know on Feb. 6, 2024. The outage was caused by a bug in the `LoadedPrograms` function that led to an infinite loop and halted consensus. see 👉 https://solana.com/news/network-performance-report-march-2024

As I recall, the bug has been found on testnet and schedule to release on mainnet but didn't make it in time then they restart with hotfix instead (rumors said that someone intend to trigger the bug manually on mainnet for some reason).

**Reply by katopz** [reply]
FYI: I do concern more about current congestion(failed tx), consider follow this https://twitter.com/laine_sa_/status/1769632485900517495 for upcoming update if you like this topic.

**Reply by tommeeeee** [reply]
Thanks you so much for information krub now I've seen more a picture of outage on Feb. 6, 2024.
for upcoming update it's really interesting krub, I'll do research about that one and contribute to book 

cause i think consensus and outage event and all about upcoming update is an important to know case every topic is associate each other. 

after consensus getting merge,  I'll create more and separate from main branch krub LOL 😊 

Now i'm gonna go to research krub, after then I'll discuss again if found some interesting information krub.

Let's get in !!! 🦀
